bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java

```diff
@@ -595,4 +597,116 @@ private boolean getConfigParameter(Map<String, Object> parameters, String parame
             return defaultValue;
         }
     }
+
+    /**
+     * For all network interfaces (except loopback) all Ipv4 addresses are returned.
+     * This list can for example, be used to scan the network for available devices.
+     * 
+     * @return A full list of IP {@link InetAddress} (except network and broadcast)
+     */
+    public static List<InetAddress> getFullRangeOfAddressesToScan() {
+        List<InetAddress> addressesToScan = List.of();
+        List<CidrAddress> ipV4InterfaceAddresses = NetUtil.getAllInterfaceAddresses().stream()
+                .filter(a -> a.getAddress() instanceof Inet4Address).collect(Collectors.toList());
+
+        for (CidrAddress i : ipV4InterfaceAddresses) {
+            addressesToScan.addAll(getAddressesRangeByCidrAddress(i, 24));
```

Comment:
Why use hard coded value of 24?  getAllinterfaceAddresses() returns ```Collection<CidrAddress>```, which allows to get the correct value for each interface using ```getPrefix()```.

Reply:
I’ll look I to this tomorrow.

Reply:
Done